Use simulate transaction jargon instead of preflight (#1381)
stellar/stellar-docs: docs/learn/encyclopedia/errors-and-debugging/debugging-errors.mdx

The typical transaction submission process can be broken down into the following sequential steps, excluding external interactions like wallet interactions. Each step has its own set of potential errors:

1. **Transaction Simulation (optional):**

>    - **What happens:** This step involves running a transaction in 'simulation' mode, which records the necessary ledger entries, CPU instructions, required authorizations, etc.

Reviewer: ```suggestion
   - **What happens:** This step involves executing the RPC endpoint [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction). The endpoint executes a transaction in 'simulation' mode of host and records the necessary ledger entries, CPU instructions, required authorizations. The end result is akin to running transactions in Core with maxed-out resources, unrestricted ledger access, and no fees.
```

>    - **Common failures:** [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction) is akin to running transactions in Core with maxed-out resources, unrestricted ledger access, and no fees. Therefore, errors typically involve exceeding the network limit or encountering contract logic issues.

Reviewer: ```suggestion
   - **Common failures:** Errors typically involve exceeding the network limit or encountering contract logic issues.
```

| --- | --- | --- |
| `$OPERATION_RESOURCE_LIMIT_EXCEEDED`, diagnostics: message specifying which resource limit has been exceeded | The transaction has exceeded the resource limit during execution. For most resources (instructions, read bytes, write bytes), this has nothing to do with the network limit; it's the transaction-specified limit. For example, a transaction has 10M instructions specified but consumes at least 10M + 1 instruction and immediately fails. Diagnostic events specify which limit has been exceeded and contain both the limit itself and the value transaction tried to consume. | If exceeding the transaction-specified limit, increase the respective resource declared in the transaction. If [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction) determined the resource limits, likely reason for exceeding is logic dependent on volatile ledger state (e.g., RNG or ledger sequence). Ensure logic is estimated sufficiently by [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction). If exceeding network limits (entry size, memory), modify/ optimize the contract. |

> | `$OPERATION_INSUFFICIENT_REFUNDABLE_FEE` | The refundable resource fee was not sufficient to cover the consumed refundable resources, i.e., not enough to pay for emitted events or TTL extensions. | The simplest fix is to increase the resource fee unconditionally; unspent fees are refunded. [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction) estimates refundable fees but can't predict TTL extensions' actual cost due to ledger state volatility. Thus [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction) can’t always predict the fee that will actually be charged. |

Reviewer 2: supernit but you went up 1 extra path here heh

```suggestion
| `$OPERATION_INSUFFICIENT_REFUNDABLE_FEE` | The refundable resource fee was not sufficient to cover the consumed refundable resources, i.e., not enough to pay for emitted events or TTL extensions. | The simplest fix is to increase the resource fee unconditionally; unspent fees are refunded. [`simulateTransaction`](../../../data/rpc/api-reference/methods/simulateTransaction) estimates refundable fees but can't predict TTL extensions' actual cost due to ledger state volatility. Thus [`simulateTransaction`](../../../data/rpc/api-reference/methods/simulateTransaction) can’t always predict the fee that will actually be charged. |
```

| `INVOKE_HOST_FUNCTION_TRAPPED`, diagnostics: HostError(Storage, LimitExceeded). | Host function tried to access a ledger entry outside of the footprint. Diagnostic events specify the exact missing entry. [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction) should usually return the valid footprint, so in the case when [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction) is being used, it’s likely that: a) Contract generates non-deterministic keys (e.g. derives them from RNG or ledger sequence), b) Contract has non-deterministic logic that results in access to a different entry set (e.g. a ‘lottery’ contract that non-deterministically performs transfer to either address A or address B). | If simualate transaction is not used, fix client code to build the correct footprint. If using simualate transaction, fix contract or footprint: avoid non-deterministic keys , and add all keys that can be accessed in any scenario, (e.g. balance entries for both A and B in the ‘lottery’ contract example). |

> | `INVOKE_HOST_FUNCTION_TRAPPED`, diagnostics: HostError(WasmVm, InvalidAction) | Failure in a Wasm contract, typically a `panic!()`. Diagnostic events might provide more detailed information, but not always, as the `panic!()` messages are not included in the Wasm builds. | Fix contract logic. Consider using `panic_with_error!()` and increase simualate transaction runs and unit testing. |

Reviewer: typo on `simualate`. Also, noticed this usage of phrasing `simulate transaction`, instead of the proper noun, [`simulateTransaction`](../../../../docs/data/rpc/api-reference/methods/simulateTransaction) as used earlier in this same page to reference the process of simulating a transaction, it seems slightly inconsistent, for consideration, the other references could just be code formatted name, `simulateTransaction`, and not hyperlinked?
